### ISSUE TITLE
Replace to this version of JS?

### ISSUE THREAD

**Issue by gitcoolcorner** [freyastreamlit/streamlit-lightweight-charts#30]
I recently came across this version of https://jsfiddle.net/adrianntf/6qea5ytv/ TV lightweight JS and this support panes outright, with no alignment issues etc as the panes support is built-in. I am not sure which version of TV LW JS you are using in this package.

And I was wondering if you could replace https://jsfiddle.net/adrianntf/6qea5ytv/ TV lightweight JS in your streamlit lightweight charts package.

**Comment by freyastreamlit** [comment]
Hi @gitcoolcorner 

This wrapper uses the official mainstream build, and it will have updates accordingly, etcetera...

But, I, myself have been using that frozen build - it is old and frozen, so it will not have any updates, bug fixes, etcetera - IT IS AS IT IS - so be careful where you go to use

Decided to make that repo Public:
- https://github.com/freyastreamlit/streamlit-lightweight-charts-ntf
and I pushed to PiPY 
- https://pypi.org/project/streamlit-lightweight-charts-ntf/

Please have a go and let me know...

Best,

Joe Rosa

**Comment by gitcoolcorner** [comment]
This is great, worked like a charm! Thanks so much. One question though:

1. In lightweight charts if I remember correctly there is an update method for the series if we want to update the charts in real time, can we use it with it?

**Comment by freyastreamlit** [comment]
HI @gitcoolcorner 

Adrian's (ntf) last 'master' merge was on 23 Feb 2022

- https://github.com/tradingview/lightweight-charts/releases

so I guess we could assume that functionality until v3.8.0 would be there.
v4 has breaking changes, so I guess he will not update from now on - but not much happened since then

So, closing this ticket - Enjoy

Best,

Joe Rosa

**Comment by gitcoolcorner** [comment]
Thanks for your prompt replies.

When I meant update method, I meant the update method that is there in the TV lightweight charts which can update or draw the last bar to make the charts realtime. We have one method which is xyzSeries.setData() and other method is xyzSeries.update(), I was wondering how would we use it with your library here: 

https://github.com/freyastreamlit/streamlit-lightweight-charts-ntf

Sorry about the confusion earlier.

**Comment by freyastreamlit** [comment]
Hi @gitcoolcorner 

Well, streamlit is not conceived for that. I know some people try a dashboard type like
https://blog.streamlit.io/how-to-build-a-real-time-live-dashboard-with-streamlit/
It is a continuous page refresh that gives the feeling is live

But that is not what streamlit is for, nor the 'lightweight charts` libraries, using setData or updateData the page still needs to re-render each time: https://tradingview.github.io/lightweight-charts/docs

If you are looking to setup your own live I would suggest:
https://medium.com/marcius-studio/financial-charts-for-your-application-cfcceb147786
That will use the real full-blown TV JS, and there are overlays and a community around...

Good luck on your projects

**Comment by gitcoolcorner** [comment]
I did look at the TradingVue a couple of weeks back, but the the thing is I am very new to the whole world of JS and I could not even get it to work (the basic, may be issue with my nodejs/npm/yarn etc). That is how I came across your repo.

I will however keep at it and see what I could do with it to resolve the JS environment issues. Meanwhile I can use lightweight charts with Streamlit (your lib) https://github.com/freyastreamlit/streamlit-lightweight-charts-ntf

I really appreciate all the help. Thanks again.
